### PR TITLE
Move regex validation checks to template schema

### PR DESCRIPTION
**Pull request by shreyas-goenka**
## Changes
This PR moves regex validation checks to the template schema. This has the following advantages:
1. Mlops-stacks will fail / give feedback early when an invalid value is provided
2. Allows us to remove existing validation checks 

Note: We first need to get https://github.com/databricks/cli/pull/912 in, and set / update the minimum CLI version for mlops-stacks before this PR can be merged.

## Tests

Tested manually. Here are the cases:

Case project_name: ab
```
shreyas.goenka@THW32HFW6T playground % cli bundle init ~/mlops-stack
Welcome to MLOps Stacks. For detailed information on project generation, see the README at https://github.com/databricks/mlops-stacks/blob/main/README.md. 

Project Name [my-mlops-project]: ab
Error: invalid value for input_project_name: "ab". Project name must be at least 3 characters long and cannot contain the following characters: "\", "/", " " and ".".
```

Case project_name: abcd/
```
shreyas.goenka@THW32HFW6T playground % cli bundle init ~/mlops-stack
Welcome to MLOps Stacks. For detailed information on project generation, see the README at https://github.com/databricks/mlops-stacks/blob/main/README.md. 

Project Name [my-mlops-project]: abcd/
Error: invalid value for input_project_name: "abcd/". Project name must be at least 3 characters long and cannot contain the following characters: ",", "/", " " and ".".
```

Case staging URL: abc
```
URL of staging Databricks workspace, used to run CI tests on PRs and preview config changes before they're deployed to production. Default: 
Azure - https://adb-xxxx.xx.azuredatabricks.net
AWS - https://your-staging-workspace.cloud.databricks.com
: abc
Error: invalid value for input_databricks_staging_workspace_host: "abc". Databricks staging workspace host URLs must start with https. Got invalid workspace host.
```

### CODE REVIEW

**Review by arpitjasa-db — COMMENTED** [review]
This is awesome thanks a ton @shreyas-goenka ! I'll approve once the CLI PR merges and we can test this out with some project generations

File: library/input_validation.tmpl
Comment (shreyas-goenka) [reply library/input_validation.tmpl]: Good point, removed! We also no longer need this variable (`$project_name_alphanumeric_underscore`) here since it was only used for validation.

**Review by arpitjasa-db — COMMENTED** [review]
@shreyas-goenka would you mind updating the tests in `tests/test_create_project.py::test_generate_fails_with_invalid_params` for the expected error now? I believe the tests should pass after that and we can merge this in!

**Comment by shreyas-goenka** [comment]
This PR requires a CLI side fix before proceeding: https://github.com/databricks/cli/pull/959

**Comment by shreyas-goenka** [comment]
TODO: update install.sh version

File: databricks_template_schema.json
Comment (arpitjasa-db) [inline databricks_template_schema.json:3]: Would you mind updating the main README to `v0.209.1` as well?

**Review by arpitjasa-db — APPROVED** [review]
Approved with one minor ask, thanks @shreyas-goenka !

File: databricks_template_schema.json
Comment (shreyas-goenka) [reply databricks_template_schema.json:3]: done!

**Review by arpitjasa-db — APPROVED** [review]

**Pull request merged**
